### PR TITLE
[MGDOBR-402] Add appsre script to build and push fleet manager image

### PR DESCRIPTION
**Pull request by msmagnanijr**
Signed-off-by: Mauricio Magnani <msmagnanijr@gmail.com>

[MGDOBR-402](https://issues.redhat.com/browse/MGDOBR-402) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

* <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

* <b>Rebase the pull request</b>  
  Comment with: `/rebase`.

</details>

### CODE REVIEW

File: hack/Makefile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.PHONY: build push
```
Comment (danielezonca) [inline hack/Makefile:1]: Is there any requirement to use makefile for this? All our scripts are for now plain bash

Comment (msmagnanijr) [reply hack/Makefile:1]: not really, I'm just following the pattern used in the AppSRE repos, but that's not required.

Comment (msmagnanijr) [reply hack/Makefile:1]: @danielezonca do I need to change this or can we leave it for now?

File: hack/Makefile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+IMAGE_NAME := quay.io/app-sre/fleet-manager
```
Comment (robpblake) [inline hack/Makefile:3]: @msmagnanijr Do you need to change the image tag to be specific to our project e.g quay.io/app-sre/rhose-fleet-manager ?

File: hack/Makefile (hunk 1 above)
Comment (robpblake) [reply hack/Makefile:1]: I think we should go for `bash` unless that's not supported by app-interface? We've already got a lot of bash in the project, so we could create a `build` or `app-interface` directory with specific scripts in there to support the builds.

**Review by robpblake — COMMENTED** [review]
@msmagnanijr Is the `hack` directory mandated or can we be a bit more explanatory with the name e.g `app-interface`?

File: hack/Makefile (hunk 1 above)
Comment (danielezonca) [reply hack/Makefile:1]: Ok let's keep it as a normal bash script for now and if necessary we can easily create the makefile.
@robpblake 
Is it fine for you?

File: hack/Makefile (hunk 2 above)
Comment (msmagnanijr) [reply hack/Makefile:3]: @robpblake it makes sense. Let me update it :)

Comment (msmagnanijr) [reply hack/Makefile:3]: just to be clear, our images are required to be in org appsre no quay.

File: hack/Makefile (hunk 1 above)
Comment (robpblake) [reply hack/Makefile:1]: Yep, +1 to bash. There is a lot of bash skills on the team, so that helps us even more.

**Comment by msmagnanijr** [comment]
> @msmagnanijr Is the `hack` directory mandated or can we be a bit more explanatory with the name e.g `app-interface`?

it is not mandatory. I'll update to "app interface".

**Review by robpblake — APPROVED** [review]

**Review by r00ta — APPROVED** [review]

**Pull request merged**